chain-signatures/contract/src/lib.rs

```diff
@@ -94,6 +95,62 @@ pub struct YieldIndex {
     data_id: CryptoHash,
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum SignatureError {
+    #[error("Signature request has timed out.")]
+    Timeout,
+    #[error("This sign request was removed from pending requests: timed out or completed.")]
+    RequestNotInPending,
+    #[error("Signature could not be verified.")]
+    SignatureNotVerified,
```

Comment:
Can we please separate errors by function or scope where they can appear?
The end goal here is to give users a finite number of errors that can happen for each call they are making.

In this case, we have `Timeout`, which the user can get from `sign()` and `SignatureNotVerified`, which can only happen in `respond`.

Reply:
sounds good. Will do.